data-structures/dictionaries_new/python/nested_dictionaries_find.py

```diff
@@ -0,0 +1,35 @@
+# Raspberry Pi Foundation
+# Developed as part of Ada Computer Science
+# 
+# Usage licensed under CC BY-NC-SA 4.0
+    
+
+def main():
+    # Declare and initialise the nested dictionary of astronauts
+    astronauts_by_name = {
+        "Sergey Prokopyev": {"craft": "ISS", "days": "176"},
+        "Dmitry Petelin": {"craft": "ISS", "days": "176"},
+        "Frank Rubio": {"craft": "ISS", "days": "176"},
+        "Fei Junlong": {"craft": "Shenzhou 15", "days": "112"},
+        "Deng Qingming": {"craft": "Shenzhou 15", "days": "107"},
+        "Zhang Lu": {"craft": "Shenzhou 15", "days": "107"},
+        "Stephen Bowen": {"craft": "ISS", "days": "54"},
+        "Warren Hoburg": {"craft": "ISS", "days": "14"},
+        "Sultan Al Neyadi": {"craft": "ISS", "days": "14"},
+        "Andrey Fedyaev": {"craft": "ISS", "days": "14"}
+    }
+
+    # Ask the user for a key to find
+    input_key = input("Enter a key: ")
```

Comment:
should we say "Enter a key (astronaut name): ")

Reply:
I need to read content but is the emphasis on the key?  If not I suggest we just say "Enter an astronaut name: "

and in the next bit saying astronaut found or not found (or even use the actual name entered).